javascript/javascript1/week4/voice-assistant/voice-assistant.js

```diff
@@ -0,0 +1,93 @@
+const dayjs = require('dayjs');
+const math = require('mathjs');
+
+function getReply(command) {
+  const storeName = "Benjamin";
+  const toDo = ["fishing", "singing in the shower"];
+  const myDate = dayjs();
+  const formatOfDate = myDate.format("dddd DD of MMMM YYYY, hh:mm A");
+  
+  switch (command) {
+    case `Hello my name is ${storeName}`:
+      return `Nice to meet you ${storeName}`;
+
+    case "What is my name?":
+      if (storeName.toLowerCase) {
+        return `Your name is ${storeName}`;
+      } else {
+        return "Your name has not been set yet";
+      }
+
+    case "What day is it today?":
+      return `Today is ${formatOfDate}`;
+
+    default:
+      if (command.startsWith("Add ")) {
+        const task = command.replace("Add ", "").replace(" to my todo", "");
+        toDo.push(task);
+        return `${task} added to your todo`;
+      }
+
+      if (command.startsWith("Remove ")) {
+        const task = command
+          .replace("Remove ", "")
+          .replace(" from my todo", "");
+        const index = toDo.indexOf(task);
+        if (index > -1) {
+          toDo.splice(index, 1);
+          return `Removed ${task} from your todo`;
+        } else {
+          return `${task} not found in your todo`;
+        }
+      }
+
+      if (command === "What is on my todo?") {
```

Comment:
I can see you have taken the approach of checking if the strings `===` or `.startsWith` a value and then `.replace`. This is not incorrect but I can recommend you attempting this again throughout your code by using regex - it's a good thing to learn and a big shortcut for the future!

An example of how the checks could be done are as so:

```
const patterns = {
  namePattern: /Hello, my name is (.+)/i,
  addPattern: /Add (.+) to my to do list\./i,
  removePattern: /Remove (.+) from my to do list\./i,
  showToDoPattern: /What is my to do\?/,
  mathPattern: /what is (\d+)\s*([+\-*\/])\s*(\d+)\?/i,
  tempPattern: /It is (\d+) degrees. What do I wear\?/,
  getNamePattern: /What is my name\?/i,
  datePattern: /What day is it today\?/i,
  timerPattern: /Set a timer for (\d+) min/i,
};
```

Then conditionally `.test` to see if it matches as a `case` and then execute that condition.

You can [read more about regex here](https://www.w3schools.com/jsref/jsref_obj_regexp.asp)

It will require you Google each individual regex pattern but it comes in handy in the future!